### PR TITLE
Allow to global turn off/on actionable diagnostic

### PR DESCRIPTION
**Pull request by lwronski**

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/bsp/BspImpl.scala [hunk 1]
```diff
@@ -149,9 +151,18 @@ final class BspImpl(
       buildChangedTest
     )
 
+    if (actionableDiagnostics.getOrElse(false))
+      logActionableDiagnostics(options0Test.orElse(options0Main), persistentLogger)
+
     PreBuildProject(mainScope, testScope, persistentLogger.diagnostics)
   }
 
+  private def logActionableDiagnostics(projectOptions: BuildOptions, logger: Logger): Unit = {
```
Comment (Gedochao) [inline modules/build/src/main/scala/scala/build/bsp/BspImpl.scala]: NIT This method is duplicated in this PR (the other instance is in `Build.scala`)
I wonder if this logic could become an extension for `BuildOptions`?

File: modules/build/src/main/scala/scala/build/Build.scala [hunk 2]
```diff
@@ -366,9 +369,19 @@ object Build {
     for (testBuild <- builds.get(Scope.Test))
       copyResourceToClassesDir(testBuild)
 
+    if (actionableDiagnostics.getOrElse(false))
+      logActionableDiagnostics(builds, logger)
+
     builds
   }
 
+  private def logActionableDiagnostics(builds: Builds, logger: Logger): Unit = {
```
Comment (Gedochao) [inline modules/build/src/main/scala/scala/build/Build.scala]: NIT (the duplicated code I mention in `BspImpl`)

**Review by Gedochao — APPROVED** [review]

File: modules/cli/src/main/scala/scala/cli/config/Keys.scala [hunk 3]
```diff
@@ -15,7 +15,10 @@ object Keys {
   val sonatypeUser     = new Key.PasswordEntry(Seq("sonatype"), "user")
   val sonatypePassword = new Key.PasswordEntry(Seq("sonatype"), "password")
 
+  val actionableDiagnostics = new Key.BooleanEntry(Seq.empty, "actionable")
```
Comment (alexarchambault) [inline modules/cli/src/main/scala/scala/cli/config/Keys.scala]: Maybe rename this to `actions`? Basically, users enable or disable suggested actions.

File: modules/options/src/main/scala/scala/build/actionable/ActionablePreprocessor.scala [hunk 4]
```diff
@@ -3,13 +3,14 @@ package scala.build.actionable
...
 import scala.build.options.BuildOptions
+import scala.build.errors.Diagnostic
 
 object ActionablePreprocessor {
   val actionableHandlers = Seq[ActionableHandler[_]](
     ActionableDependencyHandler
   )
 
-  def generateActionableDiagnostics(
+  def generateActionableDiagnostic(
```
Comment (alexarchambault) [inline modules/options/src/main/scala/scala/build/actionable/ActionablePreprocessor.scala]: NIT Why this name change?

Comment (lwronski) [reply modules/options/src/main/scala/scala/build/actionable/ActionablePreprocessor.scala]: yeah, I restore previous name

**Review by romanowski — APPROVED** [review]

**Pull request merged**